music21/musicxml/xmlToM21.py

```diff
@@ -3939,6 +3939,12 @@ def xmlNotationsToSpanners(self, mxNotations, n):
                                     )
             self.setColor(mxObj, slur)
 
+        for mxObj in mxNotations.findall('technical/hammer-on'):
+            self.xmlOneSpanner(mxObj, n, articulations.HammerOn)
+
+        for mxObj in mxNotations.findall('technical/pull-off'):
+            self.xmlOneSpanner(mxObj, n, articulations.PullOff)
+
```

Comment:
Ok I understand we might want to add them in the articulations. But we can not use `xmlTechnicalToArticulation` as they are not in TECHNICAL_MARKS anymore. So I think we would like hammer-on to be in TECHNICAL_MARKS for reading (so it would automatically benefit from the standard way to be appended to the articulations list) but not for writing (as we don't want it to be written as standard technicals). Maybe the ultimate perspective would be to have 2 distinct lists TECHNICAL_MARKS, one for reading and one for writing. But I guess this would require some important code modifications which are unnecessary at this stage.

So for now when reading I'll explicitly  append them in the articulation list which seems to be the most reasonable choice at the moment.

Reply:
Thanks, that seems like a wise course of action. When CI passes this will be mergeable, I think!

Reply:
We reversed course in 72b79c4 and decided not to put these objects in the articulations array.